File: daily/Arrays/121-Best-time-to-buy-and-sell-stock.go
package daily

func maxProfit(prices []int) int {
	left := 0
	n := len(prices)
	res := 0
	for right := 1; right < n; right++ {
		currentProfit := prices[right] - prices[left]
		if currentProfit >= 0 {
			res = max(res, currentProfit)
		} else {
			left = right
		}
	}
	return res
}

File: daily/Arrays/121-Best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        n = len(prices)
        res = 0
        for right in range(1, n):
            current_profit = prices[right] - prices[left]
            if current_profit >= 0:
                res = max(res, current_profit)
            else:
                left = right
        return res